### PR TITLE
Detect newline type from documents

### PR DESCRIPTION
**Pull request by mlange-42**

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -105,7 +113,20 @@ pub fn parse(
         return Err(msg.into());
     }
 
-    Ok((Document::new(nodes), links))
+    Ok((Document::new(nodes, newline), links))
+}
+
+fn detect_newline(text: &str) -> &'static str {
+    NEWLINE_REGEX.captures_iter(text).next().map_or_else(
```
Comment (adamreichold) [inline src/parse.rs:120]: Wouldn't it be simpler to just check `text.contains("\r\n")` and assume Unix line endings otherwise?

Comment (adamreichold) [reply src/parse.rs:120]: (The mixed cases are wrong either way, aren't they?)

File: src/document.rs [hunk 2]
```diff
@@ -6,10 +6,14 @@ use std::collections::HashMap;
 use std::fmt::Write;
 use std::path::{Path, PathBuf};
 
+pub(crate) const CRLF_NEWLINE: &str = "\r\n";
```
Comment (adamreichold) [inline src/document.rs:9]: As this is binary, i.e. leaf node in the dependency graph, `pub(crate)` and `pub` are equivalent.

File: src/parse.rs (hunk 1 above)
Comment (mlange-42) [reply src/parse.rs:120]: Yes, mixed is evil! 😈 

I wanted to simply use the first newline detected, as `text.contains("\r\n")` would use win endings as soon as any "wrong" line ending is found in a unix-style file.

I just completed to re-do everything on a separate, clearer branch, so I will move everything to another MR and close this one.

**Comment by mlange-42** [comment]
Withdrawn in favor of #122

File: src/parse.rs (hunk 1 above)
Comment (mlange-42) [reply src/parse.rs:120]: Changed as suggested, in #122

Comment (mlange-42) [reply src/parse.rs:120]: But see https://github.com/mlange-42/yarner/pull/122#discussion_r578768880

**Pull request closed without merging**